powertools-idempotency/src/main/java/software/amazon/lambda/powertools/idempotency/IdempotencyConfig.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package software.amazon.lambda.powertools.idempotency;
+
+/**
+ * Configuration of the idempotency feature. Use the {@link Builder} to create an instance.
+ */
+public class IdempotencyConfig {
+    private final int localCacheMaxItems;
+    private final boolean useLocalCache;
+    private final int expirationInSeconds;
+    private final String eventKeyJMESPath;
+    private final String payloadValidationJMESPath;
+    private final boolean throwOnNoIdempotencyKey;
+    private final String hashFunction;
+
+    private IdempotencyConfig(String eventKeyJMESPath, String payloadValidationJMESPath, boolean throwOnNoIdempotencyKey, boolean useLocalCache, int localCacheMaxItems, int expirationInSeconds, String hashFunction) {
+        this.localCacheMaxItems = localCacheMaxItems;
+        this.useLocalCache = useLocalCache;
+        this.expirationInSeconds = expirationInSeconds;
+        this.eventKeyJMESPath = eventKeyJMESPath;
+        this.payloadValidationJMESPath = payloadValidationJMESPath;
+        this.throwOnNoIdempotencyKey = throwOnNoIdempotencyKey;
+        this.hashFunction = hashFunction;
+    }
+
+    public int getLocalCacheMaxItems() {
+        return localCacheMaxItems;
+    }
+
+    public boolean useLocalCache() {
+        return useLocalCache;
+    }
+
+    public int getExpirationInSeconds() {
+        return expirationInSeconds;
+    }
+
+    public String getEventKeyJMESPath() {
+        return eventKeyJMESPath;
+    }
+
+    public String getPayloadValidationJMESPath() {
+        return payloadValidationJMESPath;
+    }
+
+    public boolean throwOnNoIdempotencyKey() {
+        return throwOnNoIdempotencyKey;
+    }
+
+    public String getHashFunction() {
+        return hashFunction;
+    }
+
+
+    /**
+     * Create a builder that can be used to configure and create a {@link IdempotencyConfig}.
+     *
+     * @return a new instance of {@link IdempotencyConfig.Builder}
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+
+        private int localCacheMaxItems = 256;
+        private boolean useLocalCache = false;
```

Comment:
Should local caching be enabled by default?

Reply:
it was not the case in python, I took the same behaviour, but I agree we should put it true as default, better performance (especially for java).

Reply:
> > Yes I had the issue, I'm on M1. It works using another JDK (x86 64-bit):
> > > java version "1.8.0_202"
> > > Java(TM) SE Runtime Environment (build 1.8.0_202-b08)
> > > Java HotSpot(TM) 64-Bit Server VM (build 25.202-b08, mixed mode)
> 
> Hmm I didn't try this specific version, But any other way not to force developing module on a specific java version for M1s ? I dont think its ideal to force a specific version. I am using latest corretto java 8. If there is no other way of configuring things differently, then may be we should re consider how the tests are written ?

I will have a look at the comments on the [issue](https://github.com/aws-samples/aws-dynamodb-examples/issues/22), see if there are alternative solutions.

Happy to hear if you have other ideas on how to test DynamoDB locally without Dynamodb Local. Docker? I thought about that initially, but I remember having issues on Github as build are already executed in docker (docker in docker is often an issue).